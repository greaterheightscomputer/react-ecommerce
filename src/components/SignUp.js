import React from 'react';
import moment from 'moment';

export default class SignUp extends React.Component {
    state = {
        email: '',
        password: '',
        firstName:'',
        lastName:'',
        phone: 0,
        createdAt: moment().toDate().valueOf()
    };
    onEmailChange = (e) => {      
        const email = e.target.value; 
        this.setState({ email });
    };
    onPasswordChange = (e) => {      
        const password = e.target.value;  
        this.setState({ password });
    };
    onFirstNameChange = (e) => {      
        const firstName = e.target.value;  
        this.setState({ firstName });
    };
    onLastNameChange = (e) => {      
        const lastName = e.target.value; 
        this.setState({ lastName });
    };
    onPhoneChange = (e) => {      
        const phone = e.target.value;  
        if (!phone || phone.match(/^\d{0,11}?$/)){
            this.setState({ phone });
        }                        
    };       
    onSubmitClick = (e) => {
        e.preventDefault();
        if(!this.state.email || !this.state.password || !this.state.firstName || !this.state.lastName || !this.state.phone) {
            this.setState(() => ({
                error: 'Please provide value for empty field(s)'
            }));
        } else {
            this.setState(() => ({ 
                error: '',
                email: this.state.email,
                password: this.state.password,
                firstName: this.state.firstName,
                lastName: this.state.lastName,
                phone: this.state.phone                                
             }));
            console.log(this.state)
        }
    }; 
    render() {
        return (
            <div>
                {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.onSubmitClick} >
                    <h1> Sign Up</h1>
                    <div>
                        <label>Email</label>
                        <input 
                            type="email"                             
                            value={this.state.email}
                            onChange={this.onEmailChange} 
                        />
                    </div>
                    <div>
                        <label>Password</label>
                        <input 
                            type="password"                            
                            value={this.state.password}
                            onChange={this.onPasswordChange} 
                        />
                    </div>
                    <div>
                        <label>First Name</label>
                        <input 
                            type="text"                             
                            value={this.state.firstName}
                            onChange={this.onFirstNameChange} 
                        />
                    </div>
                    <div>
                        <label>Last Name</label>
                        <input 
                            type="text"                             
                            value={this.state.lastName}
                            onChange={this.onLastNameChange} 
                        />
                    </div>
                    <div>
                        <label>Phone N0</label>
                        <input 
                            type="text"                            
                            value={this.state.phone}
                            onChange={this.onPhoneChange} 
                        />
                    </div>
                    <div>
                        <button>Sign Up</button>
                    </div>
                </form>
            </div>
        );
    };
};