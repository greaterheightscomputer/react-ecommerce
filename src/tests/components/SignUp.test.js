import React from 'react';
import { shallow } from 'enzyme';
import SignUp from '../../components/SignUp';

test('should render SignUp with default values', () => {
    const wrapper = shallow(<SignUp />);
    expect(wrapper).toMatchSnapshot();
});

test('should render error for invalid form submission', () => {
    const wrapper = shallow(<SignUp />);    
    wrapper.find('form').simulate('submit', {
        preventDefault: () => {}
    });
    expect(wrapper.state('error').length).toBeGreaterThan(0); 
    expect(wrapper).toMatchSnapshot();
});

test('should set email on input change', () => {
    const value = 'khadijat@g.com';
    const wrapper = shallow(<SignUp />);
    wrapper.find('input').at(0).simulate('change', {
        target: { value }
    });
    expect(wrapper.state('email')).toBe(value);
    expect(wrapper).toMatchSnapshot();
});

test('should set password on input change', () => {
    const value = '123abce';
    const wrapper = shallow(<SignUp />);
    wrapper.find('input').at(1).simulate('change', {
        target: { value }
    });
    expect(wrapper.state('password')).toBe(value);
    expect(wrapper).toMatchSnapshot();
});

test('should set firstName on input change', () => {
    const value = 'Khadijat';
    const wrapper = shallow(<SignUp />);
    wrapper.find('input').at(2).simulate('change', {
        target: { value }
    });
    expect(wrapper.state('firstName')).toBe(value);
    expect(wrapper).toMatchSnapshot();
});

test('should set lastName on input change', () => {
    const value = 'Abimbola';
    const wrapper = shallow(<SignUp />);
    wrapper.find('input').at(3).simulate('change', {
        target: { value }
    });
    expect(wrapper.state('lastName')).toBe(value);
    expect(wrapper).toMatchSnapshot();
});

test('should set phone on input change', () => {
    const value = '08037899550';
    const wrapper = shallow(<SignUp />);
    wrapper.find('input').at(4).simulate('change', {
        target: { value }
    });
    expect(wrapper.state('phone')).toBe(value);
    expect(wrapper).toMatchSnapshot();
});
