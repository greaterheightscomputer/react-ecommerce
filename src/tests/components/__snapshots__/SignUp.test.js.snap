// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SignUp with default values 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <h1>
       Sign Up
    </h1>
    <div>
      <label>
        Email
      </label>
      <input
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div>
      <label>
        Password
      </label>
      <input
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div>
      <label>
        First Name
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Last Name
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Phone N0
      </label>
      <input
        onChange={[Function]}
        type="text"
        value={0}
      />
    </div>
    <div>
      <button>
        Sign Up
      </button>
    </div>
  </form>
</div>
`;

exports[`should render error for invalid form submission 1`] = `
<div>
  <p>
    Please provide value for empty field(s)
  </p>
  <form
    onSubmit={[Function]}
  >
    <h1>
       Sign Up
    </h1>
    <div>
      <label>
        Email
      </label>
      <input
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div>
      <label>
        Password
      </label>
      <input
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div>
      <label>
        First Name
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Last Name
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Phone N0
      </label>
      <input
        onChange={[Function]}
        type="text"
        value={0}
      />
    </div>
    <div>
      <button>
        Sign Up
      </button>
    </div>
  </form>
</div>
`;

exports[`should set email on input change 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <h1>
       Sign Up
    </h1>
    <div>
      <label>
        Email
      </label>
      <input
        onChange={[Function]}
        type="email"
        value="khadijat@g.com"
      />
    </div>
    <div>
      <label>
        Password
      </label>
      <input
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div>
      <label>
        First Name
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Last Name
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Phone N0
      </label>
      <input
        onChange={[Function]}
        type="text"
        value={0}
      />
    </div>
    <div>
      <button>
        Sign Up
      </button>
    </div>
  </form>
</div>
`;

exports[`should set firstName on input change 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <h1>
       Sign Up
    </h1>
    <div>
      <label>
        Email
      </label>
      <input
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div>
      <label>
        Password
      </label>
      <input
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div>
      <label>
        First Name
      </label>
      <input
        onChange={[Function]}
        type="text"
        value="Khadijat"
      />
    </div>
    <div>
      <label>
        Last Name
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Phone N0
      </label>
      <input
        onChange={[Function]}
        type="text"
        value={0}
      />
    </div>
    <div>
      <button>
        Sign Up
      </button>
    </div>
  </form>
</div>
`;

exports[`should set lastName on input change 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <h1>
       Sign Up
    </h1>
    <div>
      <label>
        Email
      </label>
      <input
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div>
      <label>
        Password
      </label>
      <input
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div>
      <label>
        First Name
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Last Name
      </label>
      <input
        onChange={[Function]}
        type="text"
        value="Abimbola"
      />
    </div>
    <div>
      <label>
        Phone N0
      </label>
      <input
        onChange={[Function]}
        type="text"
        value={0}
      />
    </div>
    <div>
      <button>
        Sign Up
      </button>
    </div>
  </form>
</div>
`;

exports[`should set password on input change 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <h1>
       Sign Up
    </h1>
    <div>
      <label>
        Email
      </label>
      <input
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div>
      <label>
        Password
      </label>
      <input
        onChange={[Function]}
        type="password"
        value="123abce"
      />
    </div>
    <div>
      <label>
        First Name
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Last Name
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Phone N0
      </label>
      <input
        onChange={[Function]}
        type="text"
        value={0}
      />
    </div>
    <div>
      <button>
        Sign Up
      </button>
    </div>
  </form>
</div>
`;

exports[`should set phone on input change 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <h1>
       Sign Up
    </h1>
    <div>
      <label>
        Email
      </label>
      <input
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div>
      <label>
        Password
      </label>
      <input
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div>
      <label>
        First Name
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Last Name
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Phone N0
      </label>
      <input
        onChange={[Function]}
        type="text"
        value="08037899550"
      />
    </div>
    <div>
      <button>
        Sign Up
      </button>
    </div>
  </form>
</div>
`;
